iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/digitaltwin/DigitalTwinClientTests.java

```diff
@@ -217,13 +248,30 @@ public void getDigitalTwinWithAzureSasCredential() {
         }
 
         // arrange
-        IotHubConnectionString iotHubConnectionStringObj =
-                IotHubConnectionStringBuilder.createIotHubConnectionString(IOTHUB_CONNECTION_STRING);
+        digitalTwinClient = buildDigitalTwinClientWithAzureSasCredential();
 
-        IotHubServiceSasToken serviceSasToken = new IotHubServiceSasToken(iotHubConnectionStringObj);
-        AzureSasCredential sasCredential = new AzureSasCredential(serviceSasToken.toString());
+        // act
+        BasicDigitalTwin response = digitalTwinClient.getDigitalTwin(deviceId, BasicDigitalTwin.class);
```

Comment:
where does this device get created?

Reply:
There is a method that executes before every test method that creates a device for the test to use. It's above in the method annotated with the "@ before"